### ISSUE TITLE
IVirtualMachineScaleSet.Capacity and IWithCapacity.WithCapacity type mismatch (long/int)

### ISSUE THREAD

**Issue by mjrousos** [Azure/azure-sdk-for-net#3448]
There's a mismatch between the data type for VMSS capacity returned by `Microsoft.Azure.Management.Compute.Fluent.IVirtualMachineScaleSet.Capacity` (long) and the type accepted by `Microsoft.Azure.Management.Compute.Fluent.VirtualMachineScaleSet.Update.IWithCapacity.WithCapacity` (int) even though conceptually these are the same piece of data and it's reasonable for customers to round-trip data between them.

Probably they both should be ints (I don't expect a VMSS capacity to really need the range of a long :)) so that code like this could drop the otherwise unnecessary cast in the last line:

```C#
var scaleSet = AzureClient.VirtualMachineScaleSets.GetById(ScaleSetId);
var newCapacity = Math.Min(MaximumNodeCount, scaleSet.Capacity + 1);
scaleSet.Update().WithCapacity((int)newCapacity).Apply(); 
```

**Comment by anuchandy** [comment]
@mjrousos thanks for catching this. Will fix it. Originally in the api spec, this field is a Long but agree with you, capacity doesn't need range of a long.

**Comment by anuchandy** [comment]
This [PR](https://github.com/Azure/azure-sdk-for-net/pull/3465) includes fix for this.

**Comment by martinsawicki** [comment]
should have been addressed in v1.2
